openedx/features/philu_courseware/models.py

```diff
@@ -1,12 +1,19 @@
+from django.contrib.auth.models import User
 from django.db import models
 from model_utils.models import TimeStampedModel
-from django.contrib.auth.models import User
-
 from opaque_keys.edx.django.models import UsageKeyField
 
 from .constants import COMPETENCY_ASSESSMENT_TYPE_CHOICES, CORRECTNESS_CHOICES
 
 
+class CompetencyAssessmentManager(models.Manager):
+    def revert_user_post_assessment_attempts(self, user, problem_id):
+        post_assessment_records = self.get_queryset().filter(problem_id=problem_id, user=user, assessment_type='post')
+        delete_result = post_assessment_records.delete()
```

Comment:
How about we put a length check here just to be sure that record exists?

Reply:
Actually this would be an extra check, as if `post_assessment_records` will have no records it won't delete anything but yes return number of deleted records that would be `0` with some other information.